packages/cashier-v2/src/lib/PaymentAgent/provider/PaymentAgentProvider.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { usePaymentAgentList } from '@deriv/api-v2';
+import type { THooks } from '../../../hooks/types';
+import { shuffleArray } from '../../../utils';
+
+type TPaymentAgentContext = {
+    isPaymentAgentListLoading: boolean;
+    isSearchLoading: boolean;
+    onChangeSearchTermHandler: (value: string) => void;
+    onSelectPaymentMethodHandler: (value: React.SyntheticEvent<HTMLInputElement, Event> | string) => void;
+    paymentAgentList?: THooks.PaymentAgentList;
+    selectedPaymentMethod?: string;
+    supportedPaymentMethods: string[];
+};
+
+const PaymentAgentContext = createContext<TPaymentAgentContext | null>(null);
+
+export const usePaymentAgentContext = () => {
+    const context = useContext(PaymentAgentContext);
+
+    if (!context)
+        throw new Error('usePaymentAgentContext() must be called within a component wrapped in PaymentAgentProvider.');
+
+    return context;
+};
+
+const defaultPaymentMethod = '0';
+
+const PaymentAgentProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
+    const { data: list, isLoading: isPaymentAgentListLoading } = usePaymentAgentList();
+    const allPaymentAgentList = useMemo(() => shuffleArray(list), [list]);
```

Comment:
Why do we shuffle this? We end up sorting it anyway at the end?

Reply:
hey @markwylde-deriv 
we are sorting payment methods for dropdown, so when the user opens dropdown he/she can see payment methods sorted alphabetically
from other hand we are shuffling payment agent cards itself. this requirement was because some of the PA's can name themselves using such characters  as "0 ! @" in the beginning of the name thus ending up in the first place on the list, in order to prevent that we are shuffling the array of payment agents